### PR TITLE
Add instance page

### PR DESCRIPTION
**Pull request by micahmo**
## Pull Request Description

This PR introduces a _very simple_ instance page. Currently it is only accessible from the post long-press menu and essentially only displays the instance heading and sidebar. It does also have the option to open the instance in the browser.

This should provide a groundwork for many other things such as...
 - a place to block an instance
 - the ability to open this page from many other places (e.g., account switcher)
 - seeing a list of communities on an instance

## Issue Being Fixed

Issue Number: N/A

## Screenshots / Recordings

https://github.com/thunder-app/thunder/assets/7417301/5cb4b339-c08b-4bec-93bf-f25f6d0d87bc

## Checklist

- [ ] Did you update CHANGELOG.md?
- [ ] Did you use localized strings where applicable?
- [ ] Did you add `semanticLabel`s where applicable for accessibility?

### CODE REVIEW

File: lib/instance/instance_view.dart [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import 'package:cached_network_image/cached_network_image.dart';
+import 'package:flutter/material.dart';
+import 'package:lemmy_api_client/v3.dart';
+import 'package:thunder/shared/common_markdown_body.dart';
+
+class InstanceView extends StatelessWidget {
+  final Site site;
+  final String? alternateSiteName;
```
Comment (hjiangsu) [inline lib/instance/instance_view.dart:8]: Out of curiosity, why is this needed?

**Review by hjiangsu — COMMENTED** [review]
Good job overall, I just had a few comments/feedback on some minor points!

File: lib/instance/instance_page.dart [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      child: SafeArea(
+        top: false,
+        child: CustomScrollView(
+          slivers: <Widget>[
+            SliverAppBar(
+              pinned: true,
+              title: Text(fetchInstanceNameFromUrl(widget.site.actorId) ?? ''),
+              actions: [
+                IconButton(
+                  tooltip: AppLocalizations.of(context)!.openInBrowser,
```
[the remark this answers is not in the document]
Comment (micahmo) [reply lib/instance/instance_page.dart]: I like it! Done in this file in f7890730de58c6ff45582f233e2fd86e7ab1aff1.

File: lib/l10n/app_es.arb [hunk 3]
```diff
@@ -112,6 +112,8 @@
   "editComment": "Editar comentario",
   "comment": "Comentario",
   "comments": "Comments",
+  "visitInstance": "Visit Instance",
```
[the remark this answers is not in the document]
Comment (micahmo) [reply lib/l10n/app_es.arb]: Done in e3978d90f69d5c2cecd3af11717cf7da8ea8e14d.

File: lib/utils/navigate_instance.dart [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  try {
+    fullSiteView = await LemmyApiV3(instanceHost).run(const GetSite()).timeout(const Duration(seconds: 5));
+  } catch (e) {}
+
+  if (fullSiteView?.siteView?.site != null && context.mounted) {
+    Navigator.of(context).push(
+      SwipeablePageRoute(
+        transitionDuration: reduceAnimations ? const Duration(milliseconds: 100) : null,
+        backGestureDetectionWidth: 45,
+        canOnlySwipeFromEdge: disableFullPageSwipe(isUserLoggedIn: authBloc.state.isLoggedIn, state: thunderBloc.state, isFeedPage: true),
```
[the remark this answers is not in the document]
Comment (micahmo) [reply lib/utils/navigate_instance.dart]: Sounds good! Done in cbaa4a1c3066adac4cbbf6b2e21f69f8c13e3d9e.

File: lib/instance/instance_view.dart (hunk 1 above)
Comment (micahmo) [reply lib/instance/instance_view.dart:8]: I was trying to interpret some strange logic in the existing community sidebar.

https://github.com/thunder-app/thunder/blob/1047f0663eff97368a7da247634a712f605106d3/lib/community/widgets/community_sidebar.dart#L144-L150

I think I interpreted this as using one of the moderator names as an alternate name for the community if needed, but I think it's actually just using the first letter as the icon if there is no icon. This is still strange to me.

I'm going to take this out and use the first letter of the community name instead!

File: lib/l10n/app_es.arb (hunk 3 above)
[the remark this answers is not in the document]
Comment (hjiangsu) [reply lib/l10n/app_es.arb]: I would same the same applies to `lib/l10n/app_pl.arb` as well! Also, since `sv.arb` and `fi.arb` files have been removed, we can also remove them from there too.

[the remark this answers is not in the document]
Comment (micahmo) [reply lib/l10n/app_es.arb]: Oops, sorry! Should be all set now.

**Review by hjiangsu — APPROVED** [review]
